### PR TITLE
*: unify the root key path joining

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Close https://github.com/tikv/pd/issues/4284.

### What is changed and how it works?

Use `server/storage/endpoint/key_path.go` to manage the key path joining.

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -647,6 +650,10 @@ func (s *Server) bootstrapCluster(req *pdpb.BootstrapRequest) (*pdpb.BootstrapRe
 	}, nil
 }
 
+func (s *Server) appendToRootPath(key string) string {
```
Comment (rleungx) [inline server/server.go:653]: How about moving it to `key_path.go`?

File: Makefile [hunk 2]
```diff
@@ -156,7 +156,6 @@ static: install-tools
 	@revive -formatter friendly -config revive.toml $(PACKAGES)
 
 tidy:
-	@echo "go mod tidy"
```
Comment (rleungx) [inline Makefile:159]: Why remove it?

Comment (JmPotato) [reply Makefile:159]: The cmd `go mod tidy` itself will be outputted if there is no `@` at the beginning. So the `echo` is unncessary.

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**